.github/workflows/lint.yaml

```diff
@@ -0,0 +1,36 @@
+# validate any terraform file changes
+name: Terraform lint
+
+env:
+  TERRAFORM_VERSION: "1.2.5"
+  TERRAGRUNT_VERSION: "0.38.6"
+
+on:
+  push:
+  pull_request:
+
+jobs:
+  terraform-lint:
+    name: lint
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
```

Comment:
Question: why do we need to set `fetch-depth` to 0? I would have thought that the default of 1 would be sufficient?